### PR TITLE
added new customized WorkflowStatusListener

### PR DESCRIPTION
**Pull request by Ismaley**
- new customized WorkflowStatusPublisher to publish workflows on dynoQueue 
- fix on completeWorkflow method to pass the right instance of workflow to the listener
- fix to override conductor's default interface implementations when an additional module contains a custom implementation of that same interface
- some integrated tests with the new status listener

### CODE REVIEW

File: contribs/src/main/java/com/netflix/conductor/contribs/listener/WorkflowStatusListenerPublisher.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import javax.inject.Inject;
+import java.util.Collections;
+
+
+public class WorkflowStatusListenerPublisher implements WorkflowStatusListener {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WorkflowStatusListenerPublisher.class);
+    private final QueueDAO queueDAO;
+    private final ObjectMapper objectMapper = new ObjectMapper();
```
Comment (emersonborges) [inline contribs/src/main/java/com/netflix/conductor/contribs/listener/WorkflowStatusListenerPublisher.java]: I believe there is an instance of ObjectMapper already created, it's better to inject than create another one

File: contribs/src/main/java/com/netflix/conductor/contribs/PublisherModule.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.netflix.conductor.contribs;
+
+import com.google.inject.AbstractModule;
+import com.netflix.conductor.contribs.listener.WorkflowStatusListenerPublisher;
+import com.netflix.conductor.core.execution.WorkflowStatusListener;
+
+public class PublisherModule extends AbstractModule {
```
Comment (emersonborges) [inline contribs/src/main/java/com/netflix/conductor/contribs/PublisherModule.java]: ```StatusListenerModule``` wouldn't be a better name?

File: test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        empty.setName("empty_workflow");
+        empty.setSchemaVersion(2);
+        empty.setWorkflowStatusListenerEnabled(true);
+        metadataService.registerWorkflowDef(empty);
+
+        String id = startOrLoadWorkflowExecution(empty.getName(), 1, "testWorkflowTerminatedListener", new HashMap<>(), null, null);
+
+        List<Message> callbackMessages = queueDAO.pollMessages(CALLBACK_QUEUE, 1, 200);
+        queueDAO.ack(CALLBACK_QUEUE, callbackMessages.get(0).getId());
+        JsonNode payload = mapper.readTree(callbackMessages.get(0).getPayload());
```
Comment (emersonborges) [inline test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java]: Try to deserialize to an existent class

File: test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+        assertEquals(id, callbackMessages.get(0).getId());
+        assertEquals("empty_workflow", payload.get("workflowType").asText());
+        assertEquals("testWorkflowTerminatedListener", payload.get("correlationId").asText());
+        assertEquals("COMPLETED", payload.get("status").asText());
+    }
+
+    @Test
+    public void testListenerOnCompletedWorkflow() throws IOException, InterruptedException {
+
```
Comment (emersonborges) [inline test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java]: Blank line

File: test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        List<Message> callbackMessages = queueDAO.pollMessages(CALLBACK_QUEUE, 1, 200);
+        queueDAO.ack(CALLBACK_QUEUE, callbackMessages.get(0).getId());
+        JsonNode payload = mapper.readTree(callbackMessages.get(0).getPayload());
+
+        assertEquals(id, callbackMessages.get(0).getId());
+        assertEquals(LINEAR_WORKFLOW_T1_T2, payload.get("workflowType").asText());
+        assertEquals("testWorkflowCompletedListener", payload.get("correlationId").asText());
+        assertEquals("COMPLETED", payload.get("status").asText());
+
```
Comment (emersonborges) [inline test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerPublisherIntegrationTest.java]: Blank line

File: test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerTestRunner.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ */
+package com.netflix.conductor.tests.listener;
+
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.util.Modules;
+import com.netflix.conductor.tests.utils.TestModule;
+import org.junit.runners.BlockJUnit4ClassRunner;
+
+public class StatusListenerTestRunner extends BlockJUnit4ClassRunner {
```
Comment (emersonborges) [inline test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerTestRunner.java]: Try to use an existent Runner

Comment (Ismaley) [reply test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerTestRunner.java]: the current existent runner uses the stub implementation of the workflowStatusListener, overriding my implementation on that runner could impact all the existent tests using that runner.

Comment (emersonborges) [reply test-harness/src/test/java/com/netflix/conductor/tests/listener/StatusListenerTestRunner.java]: ok

**Pull request merged**